clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -532,6 +540,7 @@ public Node provide() {
         private int tries = 0;
         private boolean aborted = false;
         private Node curNode = null;
+        private Long nextAllowedRetryMs = null;
```

Comment:
Would it be slightly simpler to use `private long nextAllowedRetryMs = 0`? In general `long` seems simpler than `Long`

Reply:
Hmm... in this case not sure whether we actually need a sentinel. What we want the earliest time after which we can send the request. And by definition this value should always exist, i.e. it won't be undefined. By default it will be 0 which means we can send request anytime.

Anyway, this is an internal variable and I don't have a strong opinion. I will let you decide this :)

Reply:
I see what you mean. That is a fair point. Ok, let's use 0.